projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/DataJobDeploymentExtension.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs.it.common;
+
+import static com.vmware.taurus.datajobs.it.common.BaseDataJobDeploymentIT.SIMPLE_JOB_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseDataJobDeploymentIT.DEPLOYMENT_ID;
+import static com.vmware.taurus.datajobs.it.common.BaseDataJobDeploymentIT.SIMPLE_JOB_NOTIFIED_EMAIL;
+import static com.vmware.taurus.datajobs.it.common.BaseDataJobDeploymentIT.SIMPLE_JOB_SCHEDULE;
+import static com.vmware.taurus.datajobs.it.common.BaseDataJobDeploymentIT.USER_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_TEAM_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.mapper;
```

Comment:
Don't we have a bit of cyclic dependency ? 

BaseDataJobDeploymentIT depends on DataJobDeploymentExtension (used in `@ExtendsWith`) and DataJobDeploymentExtension depends on BaseDataJobDeploymentIT (for the static variables.

 Why not move the static variables here. Then only BaseDataJobDeploymentIT would depend on DataJobDeploymentExtension which seems expected. This way DataJobDeploymentExtension can be used without BaseDataJobDeploymentIT and it may be less confusing to users.

Reply:
Thanks for the review Toni! Totally agree with you. I thought of these constants while I was implementing it, but in the end I left them as they are.

Reply:
I found even better way, now these variables are injected into each test as method parameters: jobName, teamName, deploymentId and username.

```java
@Test
   public void testExecutionStatusCountFieldsPresent(String jobName, String username) throws Exception {
      var expectedEndTimeLarger = OffsetDateTime.now();
      var expectedEndTimeSmaller = OffsetDateTime.now().minusDays(1);

      createDataJobExecution("testId", jobName, expectedEndTimeLarger);
      createDataJobExecution("testId2", jobName, expectedEndTimeSmaller);

      mockMvc.perform(MockMvcRequestBuilders.get(JOBS_URI)
                  .queryParam("query", getQuery(jobName))
                  .with(user(username)))
            .andExpect(status().is(200))
            .andExpect(content().contentType("application/json"))
            .andExpect(jsonPath("$.data.content[0].deployments[0].successfulExecutions").value(IsNull.nullValue()))
            .andExpect(jsonPath("$.data.content[0].deployments[0].failedExecutions").value(IsNull.nullValue()));

   }
```